src/libAtomVM/externalterm.c

```diff
@@ -53,22 +56,217 @@ term externalterm_to_term(const void *external_term, Context *ctx)
     int eterm_size;
     int heap_usage = calculate_heap_usage(external_term_buf + 1, &eterm_size, ctx);
 
-    struct ListHead *heap_fragment = malloc(heap_usage * sizeof(term) + sizeof(struct ListHead));
-    if (IS_NULL_PTR(heap_fragment)) {
```

Comment:
Apologies for the diff here.  Seems to be a git thing.  The major change is, add `use_heap_fragment` flag (1 is the previous behavior).  If 0, write to the context heap.  The big diff seems to be because of the indentation change in the if block